### PR TITLE
Improve test widthdrawal handler

### PR DESCRIPTION
**Pull request by 0xTitan**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

## What is the current behavior?

Resolves: https://github.com/keep-starknet-strange/satoru/issues/480

## What is the new behavior?

- Tests improved for create and cancel withdrawal

## Does this introduce a breaking change?

 No

## Other information

Tests list for create and cancel withdrawal :

- given_normal_conditions_when_create_withdrawal_then_works
- given_market_token_amount_equal_zero_when_create_withdrawal_then_fails
- given_fee_token_lower_than_execution_fee_conditions_when_create_withdrawal_then_fails
- given_caller_not_controller_when_create_withdrawal_then_fails
- given_normal_conditions_when_cancel_withdrawal_then_works
- given_unexisting_key_when_cancel_withdrawal_then_fails
- given_account_address_zero_when_cancel_withdrawal_then_fails
- given_market_token_equals_zero_when_cancel_withdrawal_then_fails
- given_caller_not_controller_when_simulate_execute_withdrawal_then_fails
- given_invalid_withdrawal_key_when_simulate_execute_withdrawal_then_fails

TODO. Due to issue on entryPointSelector with oracle_modules::with_oracle_prices_before :
- given_normal_condition_when_execute_withdrawal_then_works
- given_invalid_withdrawal_key_when_execute_withdrawal_then_fails

### CODE REVIEW

File: tests/exchange/test_withdrawal_handler.cairo [hunk 1]
```diff
@@ -97,6 +153,111 @@ fn given_unexisting_key_when_cancel_withdrawal_then_fails() {
...
+    // Key cleaning should be done in withdrawal_utils. We only check call here.
+    withdrawal_handler.cancel_withdrawal(withdrawal_key);
+}
+
+// This tests check withdrawal execution when caller address doesn't meet controller role
+// It calls withdrawal_handler.execute_withdrawal
+// The test expects the call to panic with the error 'unauthorized_access'.
 #[test]
 #[should_panic(expected: ('unauthorized_access',))]
 fn given_caller_not_controller_when_execute_withdrawal_then_fails() {
```
Comment (zarboq) [inline tests/exchange/test_withdrawal_handler.cairo:263]:
```suggestion
fn given_caller_not_keeper_when_execute_withdrawal_then_fails() {
```

**Review by zarboq — CHANGES_REQUESTED** [review]
Great ! when you're creating/cancelling withdrawal could you add assertions where you check that you can/cannot retrieve them from data_store

File: tests/exchange/test_withdrawal_handler.cairo (hunk 1 above)
Comment (0xTitan) [reply tests/exchange/test_withdrawal_handler.cairo:263]: Hello, thx for the review. You mean assertion in the function name or create new tests to test data_store access succes/fail ?

Comment (zarboq) [reply tests/exchange/test_withdrawal_handler.cairo:263]: For this one it's just the function name but in the top comment I meant adding assertions to already existing functions such as `given_normal_conditions_when_create_withdrawal_then_works`.

Maybe you could check that `get_withdrawal` returns you the withdrawal you just created. And for `given_normal_conditions_when_cancel_withdrawal_then_works` check that the key returns an empty withdrawal from data_store

Comment (0xTitan) [reply tests/exchange/test_withdrawal_handler.cairo:263]: Ok, i will improve tests. Thank you.

Comment (0xTitan) [reply tests/exchange/test_withdrawal_handler.cairo]: Update done. I noticed two things :

- Withdrawal datas are created with key =0 : https://github.com/keep-starknet-strange/satoru/blob/main/src/withdrawal/withdrawal_utils.cairo#L151
- I have to increased gas to 200 to be abale to cancel a withdrawal : https://github.com/keep-starknet-strange/satoru/pull/549/files#diff-03af4f4679e003d13470c2f5065b5c5742d9346d598f9b7f46c07ce1d4090dd2R177

One last thing. Sempgrep security rules are failing. I will check later but i don't see what i introducethat is breaking this rule.

Comment (zarboq) [reply tests/exchange/test_withdrawal_handler.cairo]: Great catch ! You can add the right key to the withdrawal struct before storing it.

For semgrep it's not an issue you introduced. It's global on all PRs, we're investigating the issue

Comment (0xTitan) [reply tests/exchange/test_withdrawal_handler.cairo]: Update done. Key added to withdrawal.

**Review by zarboq — APPROVED** [review]

**Comment by zarboq** [comment]
LGTM :rocket:

**Pull request merged**